src/timeseriesflattener/flattened_dataset.py

```diff
@@ -454,49 +454,68 @@ def _get_temporal_feature(
             .sort_index()
         )
 
-    def _check_dfs_have_same_lengths(self, dfs: list[pd.DataFrame]):
-        """Check that all dfs have the same length."""
-        df_lengths = 0
-
-        for feature_df in dfs:
-            if df_lengths == 0:
-                df_lengths = len(feature_df)
-            else:
-                if df_lengths != len(feature_df):
-                    raise ValueError("Dataframes are not of equal length")
-
-    def _check_dfs_have_identical_indexes(self, dfs: list[pd.DataFrame]):
+    @staticmethod
+    def _check_dfs_are_ready_for_concat(dfs: list[pd.DataFrame]):
         """Sample each df and check for identical indeces.
 
         This checks that all the dataframes are aligned before
         concatenation.
         """
-        for _ in range(5000):
-            random_index = random.randint(0, len(dfs[0]) - 1)
-            for feature_df in dfs[1:]:
-                if dfs[0].index[random_index] != feature_df.index[random_index]:
-                    raise ValueError(
-                        "Dataframes are not of identical index. Were they correctly aligned before concatenation?",
-                    )
+        base_df = dfs[0]
+        base_length = len(dfs[0])
+        n_dfs = len(dfs)
+
+        log.info(
+            "Checking that dataframes are ready for concatenation - namely that their indeces are aligned. This is a sanity check, and should not be necessary if the dataframes were correctly aligned before concatenation. However, any errors here will completely break predictions, so rather safe than sorry. Can take a while for a large number of dataframes, e.g. 2 minutes for 1_000 dataframes with 2_000_000 rows.",
+        )
+
```

Comment:
I like the honesty here, but it's probably a bit too much for an info-level message. I suggest splitting in two:

debug: "Checking that dataframes are ready for concatenation - namely that their indices are aligned. This is a sanity check, and should not be necessary if the dataframes were correctly aligned before concatenation. However, any errors here will completely break predictions, so rather safe than sorry. Can take a while for a large number of dataframes, e.g. 2 minutes for 1_000 dataframes with 2_000_000 rows."

info: "Checking alignment of dataframes - this might take a little while (~2 minutes for 1.000 dataframes with 2.000.000 rows)."